### PR TITLE
Chc & dv api updates

### PR DESCRIPTION
**Pull request by Akvannortwick**
Addresses the discussion on the backfill and spatial bias for the Change Healthcare and Doctor's Visits indicators. The content is preliminary and can be revised as necessary. 

- [x] Review and add suggestions for edits @rumackaaron @huisaddison

### CODE REVIEW

File: docs/api/covidcast-signals/doctor-visits.md [hunk 1]
```diff
@@ -35,13 +35,18 @@ several days after they occur, these signals are typically available with
 several days of lag. This means that estimates for a specific day are only
 available several days later.
 
+Note that we expect estimates available for the most recent X days to be substantially revised due to later data revisions. 
+
+As insurnace claims are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. Visit [Doctor's Visits](https://delphi.cmu.edu/blog/2020/11/05/a-syndromic-covid-19-indicator-based-on-insurance-claims-of-outpatient-visits/#backfill) for more information on backfill.
```
Comment (Akvannortwick) [inline docs/api/covidcast-signals/doctor-visits.md:40]: The "Doctor's Visits" text is filler, I was unsure what language to use to refer to the blog post.

**Comment by krivard** [comment]
The organization on these isn't quite right. In the existing documentation, you can consider paragraph 1 ("Note that because doctor’s visits may be reported...") to be a definition of lag, and paragraph 2 ("The amount of lag in reporting can vary, and not all ...") to be a definition of backfill. 

The new sentences relate these definitions to the behavior of the API:

> Note that we expect estimates available for the most recent X days to be substantially revised due to later data revisions. 

> As [...] are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 

We want to give folks a mental image of a timeline, with two points indicated on it:

```
THE PAST <----- A ---------- B -------| NOW
```

Where data between NOW and B is particularly unreliable and can change dramatically in the updates in the days to follow, data before A is relatively static and isn't expected to change with future updates, and data between A and B is reliable enough for most purposes but is expected to see modest changes in future updates.

In the new sentences, B=X and A="a couple of weeks".

So probably move the A/B sentences to after backfill is defined, and adjust the terminology to leverage the concepts explained in the definitions as much as possible. If you need more words/sentences to explain it clearly, 100% go for it, and if you want to meet to discuss in more detail, lmk

File: docs/api/covidcast-signals/chng.md [hunk 2]
```diff
@@ -62,6 +68,8 @@ upward spikes in the fraction of doctor's visits that are COVID-related around
 major holidays (e.g. Memorial Day, July 4, Labor Day, etc.). These spikes are
 not necessarily indicative of a true increase of COVID-19 in a location.
 
+Note that due to Y, estimates are not always comparable across geo locations. We are currently working on adjustments to correct this spatial bias.
```
Comment (rumackaaron) [inline docs/api/covidcast-signals/chng.md:71]:
```suggestion
Note that due to local differences in health coding practices, estimates are not always comparable across locations. We are currently working on adjustments to correct this spatial bias.
```
The intent is that doctors in different locations assign ICD codes differently. I don't know how to express this succintly, since the word "code" usually implies computer code and I don't know how well acquainted our audience is with the term ICD code.

File: docs/api/covidcast-signals/doctor-visits.md (hunk 1 above)
Comment (rumackaaron) [reply docs/api/covidcast-signals/doctor-visits.md:40]:
```suggestion
As insurance claims are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. See our [blog post](https://delphi.cmu.edu/blog/2020/11/05/a-syndromic-covid-19-indicator-based-on-insurance-claims-of-outpatient-visits/#backfill) for more information on backfill.
```

File: docs/api/covidcast-signals/chng.md (hunk 2 above)
Comment (krivard) [reply docs/api/covidcast-signals/chng.md:71]: "recordkeeping"?

**Comment by Akvannortwick** [comment]
@krivard Let me know if I'm on the right track, something I missed, or if we should meet to discuss further.

File: docs/api/covidcast-signals/chng.md [hunk 3]
```diff
@@ -43,10 +43,15 @@ available several days later.
 The amount of lag in reporting can vary, and not all visits are reported with
 the same lag. After we first report estimates for a specific date, further data
 may arrive about outpatient visits on that date. When this occurs, we issue new
-estimates for those dates. This means that a reported estimate for, say, June
+estimates for those dates to backfill any missing data. This means that a reported estimate for, say, June
 10th may first be available in the API on June 14th and subsequently revised on
 June 16th.
 
+As doctor’s visits data are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 
+Note that we expect estimates available for the most recent X days to be substantially revised due to later data revisions, estimates for past dates are more static due to these revisions having already occured.
```
Comment (krivard) [inline docs/api/covidcast-signals/chng.md:51]:
```suggestion
We expect estimates available for the most recent X days to change substantially in later data revisions. Estimates for dates more than W days in the past are expected to remain fairly static, as most major revisions have already occurred. 
```

Comment (krivard) [reply docs/api/covidcast-signals/chng.md:51]: @rumackaaron do we know what X and W are in the above yet?

File: docs/api/covidcast-signals/doctor-visits.md [hunk 4]
```diff
@@ -33,15 +33,20 @@ percentage of COVID-related doctor's visits in a given location, on a given day.
...
 
 The amount of lag in reporting can vary, and not all visits are reported with
 the same lag. After we first report estimates for a specific date, further data
 may arrive about outpatients visits on that date. When this occurs, we issue new
-estimates for those dates. This means that a reported estimate for, say, June
+estimates for those dates to backfill any missing data. This means that a reported estimate for, say, June
 10th may first be available in the API on June 14th and subsequently revised on
 June 16th.
 
+ As insurnace claims are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks.
```
Comment (krivard) [inline docs/api/covidcast-signals/doctor-visits.md:45]:
```suggestion
 As insurance claims are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 
```

File: docs/api/covidcast-signals/doctor-visits.md [hunk 5]
```diff
@@ -33,15 +33,20 @@ percentage of COVID-related doctor's visits in a given location, on a given day.
...
 The amount of lag in reporting can vary, and not all visits are reported with
 the same lag. After we first report estimates for a specific date, further data
 may arrive about outpatients visits on that date. When this occurs, we issue new
-estimates for those dates. This means that a reported estimate for, say, June
+estimates for those dates to backfill any missing data. This means that a reported estimate for, say, June
 10th may first be available in the API on June 14th and subsequently revised on
 June 16th.
 
+ As insurnace claims are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 
+ Note that we expect estimates available for the most recent X days to be substantially revised due to later data revisions, estimates for past dates are more static due to these revisions having already occured.
```
Comment (krivard) [inline docs/api/covidcast-signals/doctor-visits.md:46]:
```suggestion
We expect estimates available for the most recent X days to change substantially in later data revisions. Estimates for dates more than W days in the past are expected to remain fairly static, as most major revisions have already occurred.
```

Comment (krivard) [reply docs/api/covidcast-signals/doctor-visits.md:46]: I pulled 7 days from the blog post ("For a given date of service, we typically receive only about half of the total claims within seven days following that date.") and extended it to 14 to hedge against the arrival of bulk of the second half of the claims. I estimated 28 days based on the issue size of the doctor-visits signal, which is six weeks. If we estimate no measurable changes after six weeks, 4 weeks is probably safe to classify as "fairly static"

**Comment by krivard** [comment]
Waiting on backfill constants for the change signals

File: docs/api/covidcast-signals/chng.md [hunk 6]
```diff
@@ -43,10 +43,15 @@ available several days later.
 The amount of lag in reporting can vary, and not all visits are reported with
 the same lag. After we first report estimates for a specific date, further data
 may arrive about outpatient visits on that date. When this occurs, we issue new
-estimates for those dates. This means that a reported estimate for, say, June
+estimates for those dates to backfill any missing data. This means that a reported estimate for, say, June
 10th may first be available in the API on June 14th and subsequently revised on
 June 16th.
 
+As doctor’s visits data are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 
+We expect estimates available for the most recent X days to change substantially in later data revisions. Estimates for dates more than W days in the past are expected to remain fairly static, as most major revisions have already occurred.
```
Comment (krivard) [inline docs/api/covidcast-signals/chng.md:51]:
```suggestion
We expect estimates available for the most recent 7-14 days to change substantially in later data revisions. Estimates for dates more than 60 days in the past are expected to remain fairly static, as most major revisions have already occurred. 
```

**Review by krivard — APPROVED** [review]
👍

File: docs/api/covidcast-signals/doctor-visits.md [hunk 7]
```diff
@@ -33,15 +33,20 @@ percentage of COVID-related doctor's visits in a given location, on a given day.
...
 several days after they occur, these signals are typically available with
 several days of lag. This means that estimates for a specific day are only
-available several days later.
+available several days later. 
 
 The amount of lag in reporting can vary, and not all visits are reported with
 the same lag. After we first report estimates for a specific date, further data
 may arrive about outpatients visits on that date. When this occurs, we issue new
-estimates for those dates. This means that a reported estimate for, say, June
+estimates for those dates to backfill any missing data. This means that a reported estimate for, say, June
```
Comment (rumackaaron) [inline docs/api/covidcast-signals/doctor-visits.md:41]: We usually use "backfill" as a noun to describe the phenomenon of data revisioning. We could extend the definition of "backfill" to be a verb as well, or say something like "we issue new estimates for those dates that include the most recent data reports."

File: docs/api/covidcast-signals/doctor-visits.md [hunk 8]
```diff
@@ -33,15 +33,20 @@ percentage of COVID-related doctor's visits in a given location, on a given day.
...
 10th may first be available in the API on June 14th and subsequently revised on
 June 16th.
 
+ As insurance claims are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 
+We expect estimates available for the most recent 7-14 days to change substantially in later data revisions. Estimates for dates more than 28 days in the past are expected to remain fairly static, as most major revisions have already occurred.
```
Comment (rumackaaron) [inline docs/api/covidcast-signals/doctor-visits.md:46]: I'm not sure the same criteria for determining the "fairly static" threshold are the same for doctor-visits and chng. 94% of the chng data is in by 60 days, but the change in signal values might still be negligible before that since we are missing data from both the numerator and denominator.

Comment (krivard) [reply docs/api/covidcast-signals/doctor-visits.md:46]: that's fine -- do you have alternate figures for dv and chng we can use to give people a more consistent idea of what to expect?

Comment (krivard) [reply docs/api/covidcast-signals/doctor-visits.md:46]: @rumackaaron ping

File: docs/api/covidcast-signals/doctor-visits.md (hunk 7 above)
Comment (Akvannortwick) [inline docs/api/covidcast-signals/doctor-visits.md:41]:
```suggestion
will have newly issued estimates that include the most recent data reports. In other words a reported estimate for, June
```

See if this works better.

**Comment by krivard** [comment]
Okay I went Full Overkill and computed some reasonable-maybe statistics pulled from the API. GitHub won't let me post the notebook or the HTML output, but PDF works:

[inpatient_outpatient_backfill.pdf](https://github.com/cmu-delphi/delphi-epidata/files/5698935/inpatient_outpatient_backfill.pdf)

TL;DR I computed:
* the max issue for each day
* delta = the % difference between the value at each issue and the value at the max issue
* max lag where median delta is still more than 10% ("unstable; large revisions")
* min lag where median delta falls below 1% ("fairly static")

data source | signal | lag at 10% delta | lag at 1% delta
------------ | ----- | ------ | -----
chng | smoothed outpatient cli | (none) | 10
chng | smoothed outpatient covid | 6 | 45
doctor-visits | smoothed cli | 7 | 50
hospital-admissions | smoothed covid19 from claims | 13 | 57

**Comment by krivard** [comment]
@Akvannortwick would you fill in the above numbers into the corresponding spots? I'm not exactly sure what to do about the wildly different timescales for the two chng signals. Probably handle them separately?

we should also add a similar set of changes to the hospital admissions page.

File: docs/api/covidcast-signals/chng.md [hunk 9]
```diff
@@ -43,10 +43,15 @@ available several days later.
 The amount of lag in reporting can vary, and not all visits are reported with
 the same lag. After we first report estimates for a specific date, further data
 may arrive about outpatient visits on that date. When this occurs, we issue new
-estimates for those dates. This means that a reported estimate for, say, June
+estimates for those dates to backfill any missing data. This means that a reported estimate for, say, June
 10th may first be available in the API on June 14th and subsequently revised on
 June 16th.
 
+As doctor’s visits data are available at a significant and variable latency, the signal experiences heavy backfill with data delayed for a couple of weeks. 
+We expect estimates available for the most recent 7-14 days (using a lag median between 10%-1%) to change substantially in later data revisions. Estimates for dates more than 45 days at 1% lag in the past are expected to remain fairly static, as most major revisions have already occurred.
```
Comment (krivard) [inline docs/api/covidcast-signals/chng.md:51]:
```suggestion
We expect estimates available for the most recent 4-6 days to change substantially in later data revisions (having a median delta of 10% or more). Estimates for dates more than 45 days in the past are expected to remain fairly static (having a median delta of 1% or less), as most major revisions have already occurred. 
```
